### DISCUSSION TITLE
ANE stopped working

### DISCUSSION THREAD

**Discussion by trinoo1979** [airsdk/Adobe-Runtime-Support#2223] [General]
Hello guys, Im sorry for a silly question. 
I have my project developed in Animate and compiled for Android only (as APK). I use e few ANEs in it from third party producers. Prior to AIR version 33.1.1.633 all worked fine. Now I tried two higher versions of AIR (33.1.1.929 and 50.0.0.1). Project compiles without errors, but it does not run on my devices. After exploring where the problem might be and what is inside my newly packed APKs I noticed, that it looks like the way ANEs are includes with APK file has changed. (different folder structure, some things and sources are missing)....
I understand AS3 well, but Im not the ANE creator (I bouht them), so please what are my choices now? Is the only posibility to rework and recompile all ANEs, which is not possible, as I dont even have any sources, some of the creators, which ANEs Im using, also closed their business like MyFlashLabs...)? Does it means at all, that my project is FUBAR (fucked up beyond any repair) now or is there any possibility to make it to run again even with higher versions if AIR?

**Comment by ajwfrost** [comment]
Can you please try adding an entry into the application descriptor file, within the `<android>` section, saying:
```
<BuildLegacyAPK>true</BuildLegacyAPK>
```

This should mean it builds using the older mechanism and the ANE resource handling would then work how it did earlier..

thanks

**Reply by trinoo1979** [reply]
Thanks for you quick answer and thanks again, it wooorks!

**Comment by abibik1** [comment]
Hi!

I am facing basically the same problem with ANE. It was build (by me) about 7 years ago and when I moved the app to AIR 50 look slike the ANE stopped working.

But when I try to add the
`<BuildLegacyAPK>true</BuildLegacyAPK>`

I am getting following error when building APK.:

```
NOTE: Multiple Android build-tools folders exist, choosing the latest version (33.0.1) for packaging aapt tool failed:
C:\Users\Andrew\AppData\Local\Temp\b097a1ea-f03f-40aa-95d6-40babf8e8656\savedstate-1.1.0-res\values\values.xml:3: error: Found tag id where item is expected
C:\Users\Andrew\AppData\Local\Temp\b097a1ea-f03f-40aa-95d6-40babf8e8656\lifecycle-viewmodel-2.4.1-res\values\values.xml:3: error: Found tag id where item is expected
C:\Users\Andrew\AppData\Local\Temp\b097a1ea-f03f-40aa-95d6-40babf8e8656\lifecycle-runtime-2.4.1-res\values\values.xml:3: error: Found tag id where item is expected

```
Any help would be appreciated...

**Reply by marchbold** [reply]
You can't use the legacy build method with the latest Androidx libs, so you'll need to either downgrade them or update your ane.
